### ISSUE TITLE
🐛 false positive in Astro when an import follows an import assertion import

### ISSUE THREAD

**Issue by Ivo-Evans** [webpro-nl/knip#769]
### Prerequisites

- [X] I'm using the latest version
- [X] I've read the relevant [documentation](https://knip.dev)
- [X] I've searched for [existing issues](https://github.com/webpro-nl/knip/issues?q=is%3Aissue)
- [X] I've checked the [list of known issues](https://knip.dev/reference/known-issues)
- [X] I've read the [issue reproduction guide](https://knip.dev/guides/issue-reproduction)

### Reproduction url

https://github.com/Ivo-Evans/knip-import-assertion-repro

### Reproduction access

- [X] I've made sure the reproduction is publicly accessible

### Description of the issue

In an Astro project, if an import follows a JSON import with a type assertion, it is not detected. For example, this code will result in an undetected file report:

```ts
import data from "../../config/data.json" assert { type: "json" }; // You can make knip detect the log import by removing the assert or placing the json import after the TS import
import { log } from "../dependency";
log("layout loaded with data", data);
```

On the other hand both of these work fine:

```ts
import data from "../../config/data.json";
import { log } from "../dependency";
log("layout loaded with data", data);
```

```ts
import { log } from "../dependency";
import data from "../../config/data.json" assert { type: "json" };
log("layout loaded with data", data);
```

I also tried to reproduce the issue with a simpler TypeScript repo, but everything was fine there

**Comment by webpro** [comment]
Thanks for the report, @Ivo-Evans.

Please note that the compilers are just regexes trying to extract imports, which isn't as bulletproof as using actual compilers, also see https://knip.dev/features/compilers

That said, the next release should handle the situation better.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.27.5. See [Release 5.27.5](https://github.com/webpro-nl/knip/releases/tag/5.27.5) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
